src/Analyzer.Core.UnitTests/TemplateAnalyzerTests.cs

```diff
@@ -54,6 +58,53 @@ public void AnalyzeTemplate_ValidInputValues_ReturnCorrectEvaluations(string res
             Assert.AreEqual(expectedEvaluationPassCount, evaluationsWithResults.Count(e => e.Passed));
         }
 
+        [DataTestMethod]
+        [DataRow("SimpleNestedFail.json", new int[] { 36, 43, 46, 52, 53, 54 }, DisplayName = "Simple nested template example")]
+        [DataRow("DoubleNestedFail.json", new int[] { 31, 37, 59, 53, 60,  61}, DisplayName = "Nested templates with two levels")]
+        [DataRow("InnerOuterScopeFail.json", new int[] { 49, 55, 56, 101, 107, 108, 109 }, DisplayName = "Nested template with inner and outer scope")]
+        [DataRow("ParameterPassingFail.json", new int[] { 53, 59, 62, 68, 69 }, DisplayName = "Nested template with parameters passed from parent")]
+        public void AnalyzeTemplate_ValidNestedTemplate_ReturnsExpectedEvaluations(string templateFileName, dynamic lineNumbers, string templateFilePath = null)
+        {
+            string filePath = new(Path.Combine(".", "templates", templateFileName));
+            StreamReader sr = new(filePath);
+            string template = sr.ReadToEnd();
+
+            var evaluations = templateAnalyzer.AnalyzeTemplate(template, templateFilePath: templateFilePath);
+            HashSet<int> failedEvaluationLines = new();
```

Comment:
Can you add an issue so we don't forget to fix this code repetition that we already discussed? That in multiple parts of the code we are calculating the failed lines